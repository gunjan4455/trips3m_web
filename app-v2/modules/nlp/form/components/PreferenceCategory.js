import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import RangeSlider from 'modules/shared/rangeSlider';
import styles from './PreferencesCategory.cm.scss';

class PreferenceCategory extends PureComponent {

  render() {
    const { text, subText, keyValue, min, max, labels, updateFormValues, value, valueSuffix, displayValue } = this.props;
    return (
      <div className={`at_category pl24 pr24 pt24 pb15 ${styles.bottomSpacing}`}>
        <div className="flex fw9 spaceBetween">
          <p>{text}</p>
          <span className="pfc1">{displayValue ? displayValue : value} {valueSuffix}</span>
        </div>
        <p className="f12 pfc4 mt8 pb48">{subText}</p>
        <div className="">
          <RangeSlider
            min={min}
            max={max}
            labels={labels}
            onChange={updateFormValues}
            className="nurturingSlider relative"
            value={value}
            keyValue={keyValue}
          />
        </div>
      </div>
    );
  }
}

PreferenceCategory.propTypes = {
  text: PropTypes.string,
  subText: PropTypes.string,
  keyValue: PropTypes.string.isRequired,
  min: PropTypes.number.isRequired,
  max: PropTypes.number.isRequired,
  labels: PropTypes.object.isRequired,
  updateFormValues: PropTypes.func,
  value: PropTypes.number,
  valueSuffix: PropTypes.string,
  displayValue: PropTypes.string
};

PreferenceCategory.defaultProps = {
  valueSuffix: '',
  text: '',
  subText: '',
  updateFormValues: () => {},
  value: 0,
  displayValue: null
};

export default PreferenceCategory;
