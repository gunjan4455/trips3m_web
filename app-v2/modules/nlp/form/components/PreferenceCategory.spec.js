import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import PreferenceCategory from './PreferenceCategory';

configure({ adapter: new Adapter() });
let component;
const PROPS = {
  text: 'Hotel Category',
  subText: 'Reducing hotel category will affect to a high difference in budget.',
  keyValue: 'maxHotelRating',
  min: 0,
  max: 2,
  valueSuffix: 'Star',
  labels: { 0: '0', 1: '1', 2: '2'},
  value: 3
};

const createPrefereceCategoryComponent = (props) => {
  return (<PreferenceCategory {...props} />);
};

describe('Render Preference Category ', () => {

  it('should match with the previous snapshot', () => {
    const PrefereceCategoryComponent = createPrefereceCategoryComponent(PROPS);
    component = shallow(PrefereceCategoryComponent);
    expect(component).toMatchSnapshot();
  });

  it('should match with the previous snapshot when no props are passed', () => {
    const newProps = {
      keyValue: 'maxHotelRating',
      min: 0,
      max: 2,
      labels: { 0: '0', 1: '1', 2: '2'},
    };
    const PrefereceCategoryComponent = createPrefereceCategoryComponent(newProps);
    component = shallow(PrefereceCategoryComponent);
    expect(component).toMatchSnapshot();
  });

});
