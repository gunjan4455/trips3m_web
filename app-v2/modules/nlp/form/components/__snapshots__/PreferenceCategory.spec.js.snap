// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Preference Category  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PreferenceCategory
    displayValue={null}
    keyValue="maxHotelRating"
    labels={
      Object {
        "0": "0",
        "1": "1",
        "2": "2",
      }
    }
    max={2}
    min={0}
    subText="Reducing hotel category will affect to a high difference in budget."
    text="Hotel Category"
    updateFormValues={[Function]}
    value={3}
    valueSuffix="Star"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex fw9 spaceBetween"
        >
          <p>
            Hotel Category
          </p>
          <span
            className="pfc1"
          >
            3
             
            Star
          </span>
        </div>,
        <p
          className="f12 pfc4 mt8 pb48"
        >
          Reducing hotel category will affect to a high difference in budget.
        </p>,
        <div
          className=""
        >
          <RangeSlider
            className="nurturingSlider relative"
            keyValue="maxHotelRating"
            labels={
              Object {
                "0": "0",
                "1": "1",
                "2": "2",
              }
            }
            max={2}
            min={0}
            onChange={[Function]}
            tooltip={false}
            value={3}
          />
        </div>,
      ],
      "className": "at_category pl24 pr24 pt24 pb15 bottomSpacing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Hotel Category
            </p>,
            <span
              className="pfc1"
            >
              3
               
              Star
            </span>,
          ],
          "className": "flex fw9 spaceBetween",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hotel Category",
            },
            "ref": null,
            "rendered": "Hotel Category",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                3,
                " ",
                "Star",
              ],
              "className": "pfc1",
            },
            "ref": null,
            "rendered": Array [
              3,
              " ",
              "Star",
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reducing hotel category will affect to a high difference in budget.",
          "className": "f12 pfc4 mt8 pb48",
        },
        "ref": null,
        "rendered": "Reducing hotel category will affect to a high difference in budget.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RangeSlider
            className="nurturingSlider relative"
            keyValue="maxHotelRating"
            labels={
              Object {
                "0": "0",
                "1": "1",
                "2": "2",
              }
            }
            max={2}
            min={0}
            onChange={[Function]}
            tooltip={false}
            value={3}
          />,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "nurturingSlider relative",
            "keyValue": "maxHotelRating",
            "labels": Object {
              "0": "0",
              "1": "1",
              "2": "2",
            },
            "max": 2,
            "min": 0,
            "onChange": [Function],
            "tooltip": false,
            "value": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex fw9 spaceBetween"
          >
            <p>
              Hotel Category
            </p>
            <span
              className="pfc1"
            >
              3
               
              Star
            </span>
          </div>,
          <p
            className="f12 pfc4 mt8 pb48"
          >
            Reducing hotel category will affect to a high difference in budget.
          </p>,
          <div
            className=""
          >
            <RangeSlider
              className="nurturingSlider relative"
              keyValue="maxHotelRating"
              labels={
                Object {
                  "0": "0",
                  "1": "1",
                  "2": "2",
                }
              }
              max={2}
              min={0}
              onChange={[Function]}
              tooltip={false}
              value={3}
            />
          </div>,
        ],
        "className": "at_category pl24 pr24 pt24 pb15 bottomSpacing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Hotel Category
              </p>,
              <span
                className="pfc1"
              >
                3
                 
                Star
              </span>,
            ],
            "className": "flex fw9 spaceBetween",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hotel Category",
              },
              "ref": null,
              "rendered": "Hotel Category",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  3,
                  " ",
                  "Star",
                ],
                "className": "pfc1",
              },
              "ref": null,
              "rendered": Array [
                3,
                " ",
                "Star",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reducing hotel category will affect to a high difference in budget.",
            "className": "f12 pfc4 mt8 pb48",
          },
          "ref": null,
          "rendered": "Reducing hotel category will affect to a high difference in budget.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RangeSlider
              className="nurturingSlider relative"
              keyValue="maxHotelRating"
              labels={
                Object {
                  "0": "0",
                  "1": "1",
                  "2": "2",
                }
              }
              max={2}
              min={0}
              onChange={[Function]}
              tooltip={false}
              value={3}
            />,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "nurturingSlider relative",
              "keyValue": "maxHotelRating",
              "labels": Object {
                "0": "0",
                "1": "1",
                "2": "2",
              },
              "max": 2,
              "min": 0,
              "onChange": [Function],
              "tooltip": false,
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Render Preference Category  should match with the previous snapshot when no props are passed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PreferenceCategory
    displayValue={null}
    keyValue="maxHotelRating"
    labels={
      Object {
        "0": "0",
        "1": "1",
        "2": "2",
      }
    }
    max={2}
    min={0}
    subText=""
    text=""
    updateFormValues={[Function]}
    value={0}
    valueSuffix=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex fw9 spaceBetween"
        >
          <p>
            
          </p>
          <span
            className="pfc1"
          >
            0
             
            
          </span>
        </div>,
        <p
          className="f12 pfc4 mt8 pb48"
        >
          
        </p>,
        <div
          className=""
        >
          <RangeSlider
            className="nurturingSlider relative"
            keyValue="maxHotelRating"
            labels={
              Object {
                "0": "0",
                "1": "1",
                "2": "2",
              }
            }
            max={2}
            min={0}
            onChange={[Function]}
            tooltip={false}
            value={0}
          />
        </div>,
      ],
      "className": "at_category pl24 pr24 pt24 pb15 bottomSpacing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              
            </p>,
            <span
              className="pfc1"
            >
              0
               
              
            </span>,
          ],
          "className": "flex fw9 spaceBetween",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                0,
                " ",
                "",
              ],
              "className": "pfc1",
            },
            "ref": null,
            "rendered": Array [
              0,
              " ",
              "",
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "f12 pfc4 mt8 pb48",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RangeSlider
            className="nurturingSlider relative"
            keyValue="maxHotelRating"
            labels={
              Object {
                "0": "0",
                "1": "1",
                "2": "2",
              }
            }
            max={2}
            min={0}
            onChange={[Function]}
            tooltip={false}
            value={0}
          />,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "nurturingSlider relative",
            "keyValue": "maxHotelRating",
            "labels": Object {
              "0": "0",
              "1": "1",
              "2": "2",
            },
            "max": 2,
            "min": 0,
            "onChange": [Function],
            "tooltip": false,
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex fw9 spaceBetween"
          >
            <p>
              
            </p>
            <span
              className="pfc1"
            >
              0
               
              
            </span>
          </div>,
          <p
            className="f12 pfc4 mt8 pb48"
          >
            
          </p>,
          <div
            className=""
          >
            <RangeSlider
              className="nurturingSlider relative"
              keyValue="maxHotelRating"
              labels={
                Object {
                  "0": "0",
                  "1": "1",
                  "2": "2",
                }
              }
              max={2}
              min={0}
              onChange={[Function]}
              tooltip={false}
              value={0}
            />
          </div>,
        ],
        "className": "at_category pl24 pr24 pt24 pb15 bottomSpacing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                
              </p>,
              <span
                className="pfc1"
              >
                0
                 
                
              </span>,
            ],
            "className": "flex fw9 spaceBetween",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  0,
                  " ",
                  "",
                ],
                "className": "pfc1",
              },
              "ref": null,
              "rendered": Array [
                0,
                " ",
                "",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "f12 pfc4 mt8 pb48",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RangeSlider
              className="nurturingSlider relative"
              keyValue="maxHotelRating"
              labels={
                Object {
                  "0": "0",
                  "1": "1",
                  "2": "2",
                }
              }
              max={2}
              min={0}
              onChange={[Function]}
              tooltip={false}
              value={0}
            />,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "nurturingSlider relative",
              "keyValue": "maxHotelRating",
              "labels": Object {
                "0": "0",
                "1": "1",
                "2": "2",
              },
              "max": 2,
              "min": 0,
              "onChange": [Function],
              "tooltip": false,
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
